mlx/traceable_base_node.py

```diff
@@ -34,7 +34,8 @@ def create_top_node(self, title, app=None):
         admon_node['classes'].append('item')
         title_node = nodes.title()
         if app:
-            title_node += self.make_internal_item_ref(app, title).children[0]
+            showcaptions = not self['nocaptions']
+            title_node += self.make_internal_item_ref(app, title, showcaptions).children[0]
         else:
             title_node += nodes.Text(title)
```

Comment:
what happens when app is not defined? still no caption?

Reply:
In case of an item-node, the `app` argument is always defined. This is done to be able to create the permalink. All other nodes leave `app` undefined. Only the item-directive can have an item ID + caption. Other directives just have a title string.

I added this `app` kwarg in #151 so I only had to change the item-node class and not the other node classes.

Reply:
The docstring states: `app (sphinx.application.Sphinx): Optional application object, needed when item ID is given to create link`
Is this sufficiently clear?